src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskReplicationMIControllerTest.java

```diff
@@ -215,6 +320,41 @@ public void user_should_claim_task_and_claim_action_recorded_in_replicate_db() {
             .body("task_assignments_list.size()", equalTo(1))
             .body("task_assignments_list.get(0).assignment_end_reason", equalTo("UNCLAIMED"));
 
+        resultHistory = restApiActions.get(
+            ENDPOINT_BEING_TESTED_HISTORY,
+            taskId,
+            caseworkerCredentials.getHeaders()
+        );
+
+        resultHistory.prettyPrint();
+        resultHistory.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("task_history_list.size()", equalTo(4));
+
+        resultReportable = restApiActions.get(
+            ENDPOINT_BEING_TESTED_REPORTABLE,
+            taskId,
+            caseworkerCredentials.getHeaders()
+        );
+
+        resultReportable.prettyPrint();
+        resultReportable.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("reportable_task_list.size()", equalTo(1))
+            .body("reportable_task_list.get(0).state", equalTo("UNASSIGNED"))
+            .body("reportable_task_list.get(0).assignee", nullValue())
+            .body("reportable_task_list.get(0).updated_by", notNullValue())
+            .body("reportable_task_list.get(0).updated", notNullValue())
+            .body("reportable_task_list.get(0).update_action", equalTo("Unclaim"))
+            .body("reportable_task_list.get(0).created_date", notNullValue())
+            .body("reportable_task_list.get(0).due_date", notNullValue())
+            .body("reportable_task_list.get(0).last_updated_date", notNullValue())
+            .body("reportable_task_list.get(0).first_assigned_date", notNullValue())
+            .body("reportable_task_list.get(0).first_assigned_date_time", notNullValue())
+            .body("reportable_task_list.get(0).wait_time_days", equalTo(0))
+            .body("reportable_task_list.get(0).wait_time", notNullValue())
+            .body("reportable_task_list.get(0).number_of_reassignments", equalTo(0));
```

Comment:
In this scenario of claim/unclaim/claim it's important to check the first_assignment_dates.  They must be the same non null vals no matter how many assignments.  Similar with wait_time.

Reply:
Should also check that other calculated fields are not changed when they shouldn't too.

Reply:
probably worth checking task_assignments while you're at it too.

Reply:
Validations added

Reply:
I can see you've added date checking for the reporting table but I can't see any for checking the dates in the reportable_tasks table.  Can we add those too please?  Assignment start time should match the reporting table claim update time etc.

Reply:
Also Added validation on Assignment Start and End Dates.